util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -87,25 +133,71 @@ public boolean acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     }
   }
 
+  /**
+   * Override this if your keys are not of type EquivalentAddressGroup.
+   * @param key Key to identify the ChildLbState
+   * @param resolvedAddresses list of addresses which include attributes
+   * @param childConfig a load balancing policy config. This field is optional.
+   * @return a fully loaded ResolvedAddresses object for the specified key
+   */
+  protected ResolvedAddresses getChildAddresses(Object key, ResolvedAddresses resolvedAddresses,
+      Object childConfig) {
+    if (key instanceof EquivalentAddressGroup) {
+      key = new Endpoint((EquivalentAddressGroup) key);
+    }
+    checkArgument(key instanceof Endpoint, "key is wrong type");
+
+    // Retrieve the non-stripped version
+    EquivalentAddressGroup eagToUse = null;
+    for (EquivalentAddressGroup currEag : resolvedAddresses.getAddresses()) {
+      if (key.equals(currEag)) {
```

Comment:
I've changed it to create an Endpoint out of currEag for the comparison, but I'm curious as to why you feel that way. 
 Since an endpoint is just a different representation of a subset of the data in an EquivalentAddressGroup why is it weird to consider them equal?

Reply:
And I see you also updated Enpoint.equals() to no longer consider any EquivalentAddressGroup to be equal.

I think this is better because it breaks a fairly universal java convention around identity and type. Type is generally considered to be an integral part of an objects identity and instances of different types to not be equal. Even if in our use right now it might not be a problem, at a minimum, this is less surprising.